src/ModelContextProtocol/Authentication/AuthorizationHelpers.cs

```diff
@@ -0,0 +1,165 @@
+using ModelContextProtocol.Types.Authentication;
+using ModelContextProtocol.Utils.Json;
+using System.Text.Json;
+
+namespace ModelContextProtocol.Authentication;
+
+/// <summary>
+/// Provides utility methods for handling authentication in MCP clients.
+/// </summary>
+public static class AuthorizationHelpers
+{
+    /// <summary>
+    /// Fetches the protected resource metadata from the provided URL.
+    /// </summary>
+    /// <param name="metadataUrl">The URL to fetch the metadata from.</param>
+    /// <param name="cancellationToken">A token to cancel the operation.</param>
+    /// <returns>The fetched ProtectedResourceMetadata, or null if it couldn't be fetched.</returns>
+    private static async Task<ProtectedResourceMetadata?> FetchProtectedResourceMetadataAsync(
+        Uri metadataUrl, 
+        CancellationToken cancellationToken = default)
+    {
+        using var httpClient = new HttpClient();
+        try
+        {
+            var response = await httpClient.GetAsync(metadataUrl, cancellationToken);
```

Comment:
This is going to end up buffering the whole response. It'd be better to use the SendAsync overload that takes an HttpRequestMessage and, importantly, and HttpCompletionOption enum... you want to specify HttpCompletionOption.ResponseHeadersRead (the default is ResponseContentRead).